### ISSUE TITLE
Add support for `eth_simulateV1`

### ISSUE THREAD

**Issue by fselmo** [ethereum/web3.py#3494]
### What feature should we add?

The endpoint `eth_simulateV1` was [recently added to geth](https://github.com/ethereum/go-ethereum/pull/27720). This cna be quite useful for users of *web3.py*. We should add support for this new method.

**Comment by malikrohail** [comment]
can i work on this?

**Comment by kclowes** [comment]
@malikrohail yep! Before merging, we'll look for both the implementation and tests (integration probably, they live [here](https://github.com/ethereum/web3.py/blob/main/web3/_utils/module_testing/eth_module.py))
